--- a/pulp-runtime/include/archi/chips/pulpissimo/memory_map.h
+++ b/pulp-runtime/include/archi/chips/pulpissimo/memory_map.h
@@ -77,5 +77,6 @@
 #define ARCHI_CLUSTER_PERIPHERALS_ADDR             ( ARCHI_CLUSTER_ADDR + ARCHI_CLUSTER_PERIPHERALS_OFFSET )
 #define ARCHI_CLUSTER_PERIPHERALS_GLOBAL_ADDR(cid) ( ARCHI_CLUSTER_GLOBAL_ADDR(cid) + ARCHI_CLUSTER_PERIPHERALS_OFFSET )
 
+#define {name.upper()}0_BASE_ADDR 0x1a400000
 
 #endif
--- a/pulp-runtime/include/pulp.h
+++ b/pulp-runtime/include/pulp.h
@@ -27,6 +27,8 @@
 #include <archi/pulp.h>
 #include <hal/pulp.h>
 #include <data/data.h>
+#include <{name}_auto.h>
+#include <{name}_driver.h>
 
 typedef enum {
   PI_FREQ_DOMAIN_FC     = 0,
--- a/pulp-runtime/rules/pulpos/targets/pulpissimo.mk
+++ b/pulp-runtime/rules/pulpos/targets/pulpissimo.mk
@@ -57,7 +57,7 @@ soc_eu/version=2
 PULP_SRCS     += kernel/fll-v$(fll/version).c
 PULP_SRCS     += kernel/freq-domains.c
 PULP_SRCS     += kernel/chips/pulpissimo/soc.c
-
+PULP_SRCS     += drivers/{name}_driver.c
 
 include $(PULPRT_HOME)/rules/pulpos/configs/default.mk
 
